src/main/resources/assets/opencomputers/loot/openos/lib/serialization.lua

```diff
@@ -1,134 +1,93 @@
-        else
-          r = "{"
-        end
-        local tk = type(k)
-        if tk == "number" and k == i then
-          i = i + 1
-          r = r .. s(v, l + 1)
-        else
-          if tk == "string" and not kw[k] and string.match(k, id) then
-            r = r .. k
-          else
-            r = r .. "[" .. s(k, l + 1) .. "]"
-          end
-          r = r .. "=" .. s(v, l + 1)
-        end
-      end
-      ts[v] = nil -- allow writing same table more than once
-      return (r or "{") .. "}"
-    else
-      if pretty then
-        return tostring(v)
-      else
-        error("unsupported type: " .. t)
-      end
-    end
-  end
-  local result = s(value, 1)
-  local limit = type(pretty) == "number" and pretty or 10
-  if pretty then
-    local truncate = 0
-    while limit > 0 and truncate do
-      truncate = string.find(result, "\n", truncate + 1, true)
-      limit = limit - 1
-    end
-    if truncate then
-      return result:sub(1, truncate) .. "..."
-    end
-  end
-  return result
+------------------------------------------------- Private methods -----------------------------------------------------------------
+
+local function doSerialize(array, text, prettyLook, indentationSymbol, oldIndentationSymbol, equalsSymbol)
+	text = {"{"}
+	table.insert(text, (prettyLook and "\n" or nil))
+	
+	for key, value in pairs(array) do
+		local keyType, valueType, stringValue = type(key), type(value), tostring(value)
+
+		if keyType == "number" or keyType == "string" then
+			table.insert(text, (prettyLook and indentationSymbol or nil))
+			table.insert(text, "[")
+			table.insert(text, (keyType == "string" and table.concat({"\"", key, "\""}) or key))
+			table.insert(text, "]")
+			table.insert(text, equalsSymbol)
+			
+			if valueType == "number" or valueType == "boolean" or valueType == "nil" then
+				table.insert(text, stringValue)
+			elseif valueType == "string" or valueType == "function" then
+				table.insert(text, "\"")
+				table.insert(text, stringValue)
+				table.insert(text, "\"")
+			elseif valueType == "table" then
+				table.insert(text, table.concat(doSerialize(value, text, prettyLook, table.concat({indentationSymbol, indentationSymbol}), table.concat({oldIndentationSymbol, indentationSymbol}), equalsSymbol)))
+			else
+				error("Unsupported table value type: " .. valueType)
+			end
+			
+			table.insert(text, ",")
+			table.insert(text, (prettyLook and "\n" or nil))
+		else
+			error("Unsupported table key type: " .. keyType)
+		end
+	end
+
+	table.remove(text, (prettyLook and #text - 1 or #text))
+	table.insert(text, (prettyLook and oldIndentationSymbol or nil))
+	table.insert(text, "}")
+	return text
+end
+
+------------------------------------------------- Public methods -----------------------------------------------------------------
+
+function serialization.serialize(array, prettyLook, indentationWidth, indentUsingTabs)
+	checkArg(1, array, "table")
```

Comment:
it is good to keep the first two args compatible with existing user scripts. That said, let's keep the name of the first argument "value" and I don't see why serialize needs to checkArg that the 1st param is a table, doSerialize doesn't care. Also, consider not using an args list for indentationWidth and indentUsingTabs, consider using a hash of args (typically called options). Why? because when order doesn't matter and all of the args are optional (i.e. can be nil) it is best a hash.